playwright-e2e/dsm/component/tissue.ts

```diff
@@ -0,0 +1,242 @@
+import {waitForResponse} from '../../utils/test-utils';
+import {expect, Locator, Page} from '@playwright/test';
+import {
+  SequencingResultsEnum,
+  SMIdEnum,
+  TissueDynamicFieldsEnum,
+  TissueTypesEnum,
+  TumorTypesEnum
+} from '../pages/tissue-information-page/enums/tissue-information-enum';
+import TextArea from 'dss/component/textarea';
+import {
+  FillDate,
+  TissueInputsMapValue
+} from '../pages/tissue-information-page/interfaces/tissue-information-interfaces';
+import Select from 'dss/component/select';
+import DatePicker from './date-picker';
+import Input from 'dss/component/input';
+import {FillTissue} from '../pages/tissue-information-page/interfaces/fill-tissue-interface';
+import {tissueInputs} from '../pages/tissue-information-page/models/tissue-inputs';
+import {InputTypeEnum} from './tabs/enums/onc-history-input-columns-enum';
+import Button from 'dss/component/button';
+import SMID from './smid';
+
+export default class Tissue {
+  private readonly SMIDModal: SMID;
+
+  constructor(private readonly page: Page, private readonly tissueIndex: number = 0) {
+    this.SMIDModal = new SMID(this.page, this.tissueIndex);
+  }
+
+  public async getFieldValue(dynamicField: TissueDynamicFieldsEnum): Promise<string> {
+    const {
+      type: inputType,
+      byText
+    } = tissueInputs.get(dynamicField) as TissueInputsMapValue;
+
+    let value: Promise<any>;
+
+    switch (inputType) {
+      case InputTypeEnum.INPUT: {
+        const inputLocator = await this.getField(dynamicField, byText);
```

Comment:
I believe `const inputLocator = await this.getField(dynamicField, byText);` can be moved outside the `swtich` block and not duplicated in every `case`. Because `byText` is `false` by default.

Reply:
Agree! Done!